test/e2e/httpproxy/ip_filtering_test.go

```diff
@@ -0,0 +1,399 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package httpproxy
+
+import (
+	"context"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"github.com/projectcontour/contour/test/e2e"
+)
+
+func testIPFilterPolicy(namespace string) {
```

Comment:
these tests will all run in the same namespace and the HTTPProxies have the same fqdn, so I fear we will get some flakiness in CI long-term due to cross-test pollution (configuration for one test is not updated when another is running causing it to fail)

could be a cause of this run of failures, but I would have to look into it more: https://github.com/projectcontour/contour/actions/runs/4579282622/jobs/8086947055?pr=5008

Reply:
Yep, it does like like (a) giving each of the root proxies a unique FQDN; and (b) giving each proxy a unique name makes these tests pass.